manifests/agent.pp

```diff
@@ -172,13 +172,19 @@
     $_package_source = undef
   }
 
+  if $facts['service_provider'] == 'systemd' {
```

Comment:
This seems very clunky. If the docs say this is what should be done, let's put a link to the docs here and a short explanation.

Reply:
I'll add the link but this is a very common pattern I've encountered. Normally this would be handled by a module like camptocamp/systemd if we defined the unit files in Puppet or by the RPM installing the unit files. 

FWIW Puppet >= 6.1.0 will auto execute `systemctl daemon-reload` before restarting a service.  I am updating PR to only execute the command for Pupppet < 6.1.0 which is a pattern in some voxpupuli modules dealing with systemd.

Puppet change: https://github.com/puppetlabs/puppet/commit/f8d5c60ddb130c6429ff12736bfdb4ae669a9fd4